### PR TITLE
Ramenhog/search page scaffold

### PR DESCRIPTION
**Pull request by ramenhog**
## Issue Number
#8 

## Purpose/Implementation Notes
* Implement the basic scaffolding for the search results page
* See "Functional tests" for functionality covered in this ticket

### TODO:
* "Back To Top" button
* Use Input component for search input
* Apply filters (waiting on endpoint modifications)
* Sample metadata fields
* Link "View Samples" button to samples section on experiment page
* Add dropdown for changing number of results per page
* Add page to dataset functionality

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

* [ ] Bugfix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Add/Remove Experiment
- [x] Changes button to "Added" message with a "Remove" button and vice versa
- [x] Updates the dataset bubble number on the "Download Dataset" button in the navbar
- [x] When user adds first experiment, a dataSet is created via endpoint and dataSetId is stored in localStorage for future reference

#### Pagination
- [x] Previous/Next should be disabled if there is no previous/next
- [x] Clicking on page number should take user to that page number in current results and scroll them back to the top
- [x] Active page should be highlighted
- [x] Jump to page should take them to the requested page number in current results
- [x] Page changes should update query params in the URL accordingly

#### Submit new query via input at the top
- [x] Should take user back to page 1 and show new search results
- [x] Query submission should update URL query params accordingly

#### Parsing URL
- [x] Loads correct results and page based on URL query params

#### Side Effects to the Download Page
- [x] Download page now updates by fetching dataSet from endpoint
- [x] Can remove by experiment
- [x] TODO: remove by species

## Screenshots
search-scaffolding

### CODE REVIEW

File: src/common/helpers.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/**
```
Comment (arielsvn) [inline src/common/helpers.js]: 🎉 great that you got JSDoc working, I'll try to use it more!

File: src/common/helpers.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * Generates a query string from a query object
+ * @param {object} queryObj
+ * @returns {string}
+ */
+export function getQueryString(queryObj) {
+  const query = [];
+  Object.keys(queryObj).forEach(key => {
+    if (queryObj[key]) query.push(`${key}=${encodeURI(queryObj[key])}`);
```
[the remark this answers is not in the document]
Comment (ramenhog) [reply src/common/helpers.js]: Good call!

File: src/components/Button/Button.scss [hunk 3]
```diff
@@ -15,6 +15,17 @@
     color: $azure;
   }
 
+  &--plain {
+    color: $azure;
+    font-size: 0.8rem;
+    border: 0;
+    border-bottom: 1px solid currentColor;
```
[the remark this answers is not in the document]
Comment (ramenhog) [reply src/components/Button/Button.scss]: it references the color that's applied to the element its being used on. super useful for button borders. https://css-tricks.com/currentcolor/

File: src/common/helpers.js (hunk 1 above)
Comment (ramenhog) [reply src/common/helpers.js]: haha had to actually install an extension... 😢

**Comment by ramenhog** [comment]
@arielsvn I tackled all the comments we talked about in person this morning. Should be good for re-review, thanks!

**Pull request merged**